pytrip/vdx.py

```diff
@@ -221,33 +233,36 @@ def read_vdx(self, path):
         fp = open(path, "r")
         content = fp.read().split('\n')
         fp.close()
+
+        self.version = self.vdx_version(content)
+
         i = 0
         n = len(content)
         header_full = False
         #        number_of_vois = 0
         while i < n:
-            line = content[i]
+            line = content[i].strip()
             if not header_full:
-                if re.match("vdx_file_version", line) is not None:
-                    self.version = line.split()[1]
-                elif re.match("all_indices_zero_based", line) is not None:
+                if line.startswith("all_indices_zero_based"):
                     self.zero_based = True
 #                TODO number_of_vois not used
-#                elif re.match("number_of_vois", line) is not None:
+#                elif "number_of_vois" in line:
 #                    number_of_vois = int(line.split()[1])
-            if re.match("voi", line) is not None:
+            if line.startswith("voi"):
                 v = Voi(line.split()[1], self.cube)
                 if self.version == "1.2":
-                    if not line.split()[5] == '0':
-                        i = v.read_vdx_old(content, i)
+                    _token = line.split()
+                    if len(_token) == 6:
+                        if _token[5] is not '0':
```

Comment:
Interesting issue:

```
Python 2.7.14 (default, Sep 20 2017, 01:25:59) 
[GCC 7.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> line = "voi voi_empty type 0 #subvoi 0"
>>> _token = line.split()
>>> _token[5] is not '0'
False
```

```
Python 3.6.3 (default, Oct 24 2017, 14:48:20) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> line = "voi voi_empty type 0 #subvoi 0"
>>> _token = line.split()
>>> _token[5] is not '0'
False
```

and finally:
```
Python 3.2.6 (default, Nov  3 2017, 18:37:10) 
[GCC 7.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> line = "voi voi_empty type 0 #subvoi 0"
>>> _token = line.split()
>>> _token[5] is not '0'
True

```

Reply:
Wow... Maybe even a bug in python 3.2?

Reply:
Python 2.7 (and also 3.6):

```
>>> x = '0'
>>> id(x)
140615444444744
>>> id(_token[5])
140615444444744
```

It looks that memory allocation is handled in a different way in python3.2 and identity comparison doesn't work as expected, in python 3.2 we get:

```
>>> x = '0'
>>> id(x)
140514548636616
>>> id(_token[5])
140514548695312
```

Reply:
Not a bug, but probably a feature. A simple fix in 518134ef85f7c681394705393947402d0b6cf6a5 should solve it